pycls/core/checkpoint.py

```diff
@@ -72,15 +70,21 @@ def save_checkpoint(model, model_ema, optimizer, epoch, test_err, ema_err):
         "optimizer_state": optimizer.state_dict(),
         "cfg": cfg.dump(),
     }
+
     # Write the checkpoint
     checkpoint_file = get_checkpoint(epoch + 1)
+    # Save checkpoints only from the main process
+    if not dist.is_main_proc():
+        return
+
     with pathmgr.open(checkpoint_file, "wb") as f:
         torch.save(checkpoint, f)
     # Store the best model and model_ema weights so far
     if not pathmgr.exists(get_checkpoint_best()):
-        pathmgr.copy(checkpoint_file, get_checkpoint_best())
+        with pathmgr.open(get_checkpoint_best(), "wb") as f:
+            torch.save(checkpoint, f)
     else:
-        with pathmgr.open(get_checkpoint_best(), "rb") as f:
+        with open(get_checkpoint_best(), "rb") as f:
```

Comment:
Should this be "pathmgr.open"

Reply:
I had an error when using pathmgr which is why I switched to open.